copier/main.py

```diff
@@ -277,14 +278,18 @@ def _external_data(self) -> LazyDict:
         Files will only be parsed lazily on 1st access. This helps avoiding
         circular dependencies when the file name also comes from a variable.
         """
-        return LazyDict(
-            **{
-                name: lambda path=path: load_answersfile_data(
-                    self.dst_path, self._render_string(path)
-                )
-                for name, path in self.template.external_data.items()
-            }
-        )
+        # Given those values are layzily rendered on 1st access then cached
+        # the phase value is irrelevant and could be misleading.
+        # As a consequence it is explicitely set to "undefined".
+        with Phase.use(Phase.UNDEFINED):
+            return LazyDict(
+                **{
+                    name: lambda path=path: load_answersfile_data(
+                        self.dst_path, self._render_string(path)
+                    )
+                    for name, path in self.template.external_data.items()
+                }
+            )
```

Comment:
This won't work because the answers file path is rendered lazily. The phase context needs to be overridden inside the lamba. But this isn't possible I'm Python AFAIK, so we could introduce a closure instead of the lamda which renders the string inside the phase context.

Could you also add a test case please?

Reply:
Done 👍🏼